snare.py

```diff
@@ -235,6 +214,37 @@ def handle_request(self, request, payload):
                     if content_type:
                         if 'text/html' in content_type:
                             content = yield from self.handle_html_content(content)
+
+        elif detection['type'] == 2:
+            payload_content = event_result['response']['message']['detection']['payload']
+            if payload_content['page']:
+                try:
+                    file_name = self.meta[payload_content['page']]['hash']
+                    content_type = self.meta[payload_content['page']]['content_type']
+                    page_path = os.path.join(self.dir, file_name)
+                    with open(page_path, encoding='utf-8') as p:
+                        content = p.read()
+                except KeyError:
+                    content = '<html><body></body></html>'
+                    content_type = 'text\html'
+
+                soup = BeautifulSoup(content, 'html.parser')
+                script_tag = soup.new_tag('div')
+                script_tag.append(BeautifulSoup(payload_content['value'], 'html.parser'))
+                soup.body.append(script_tag)
+                content = str(soup).encode()
+            else:
+                content_type = mimetypes.guess_type(payload_content['value'])[0]
+                content = payload_content['value'].encode('utf-8')
+
+            if 'headers' in payload_content:
+                for name, val in payload_content['headers']:
+                    response.add_header(name, val)
+        else:
+            response = aiohttp.Response(
+                self.writer, status=payload_content['status_code'], http_version=request.version
```

Comment:
I think if we have some problem in tanner, we should keep it private and show e.g. index page for the user, how do you think?